web/static/js/components/subjects/Subjects.jsx

```diff
@@ -0,0 +1,240 @@
+// @flow
+import {bindActionCreators} from 'redux'
+import {connect} from 'react-redux'
+import React, {Component} from 'react'
+import { NavLink } from 'react-router-dom'
+import Card from 'react-md/lib/Cards/Card'
+import DataTable from 'react-md/lib/DataTables/DataTable'
+import TableHeader from 'react-md/lib/DataTables/TableHeader'
+import TableBody from 'react-md/lib/DataTables/TableBody'
+import TablePagination from 'react-md/lib/DataTables/TablePagination'
+import TableRow from 'react-md/lib/DataTables/TableRow'
+import TableColumn from 'react-md/lib/DataTables/TableColumn'
+import Dialog from 'react-md/lib/Dialogs'
+import CircularProgress from 'react-md/lib/Progress/CircularProgress'
+
+import * as collectionActions from '../../actions/subjects'
+import * as itemActions from '../../actions/subject'
+import EmptyListing from '../EmptyListing'
+import SubjectForm from './SubjectForm'
+import SubNav from '../SubNav'
+import type { Subject, SubjectParams } from '../../types'
+
+class SubjectsList extends Component {
+  props: {
+    items: Subject[],
+    showSubjectForm: () => void,
+    onPageChange: (page: number) => void,
+    onSubjectClick: (subject: Subject) => void,
+    currentPage: ?number,
+    rowsPerPage: number,
+    count: number,
+  }
+
+  render() {
+    const subjects = this.props.items || []
+
+    if (subjects.length == 0) {
```

Comment:
Won't this display "You have no subjects on this campaign" while loading the list?

Reply:
That's the behaviour for Campaigns. But here, [when the app is fetching, we don't even render a `SubjectsList`](https://github.com/instedd/active-monitoring/pull/58/files/4989e22ad637f9d5649b4509580df9eaa5769736#diff-fe2b91a450fd53a2d1b4f0678075545bR206).

So - no, we're not rendering that while loading.